### PR TITLE
feat: transaction cancel / speed up.

### PR DESCRIPTION
**Pull request by liuzjalex**

### CODE REVIEW

**Review by sidmorizon — DISMISSED** [review]

**Review by ByteZhang1024 — DISMISSED** [review]

File: packages/engine/src/index.ts [hunk 1]
```diff
@@ -93,6 +93,13 @@ import {
   DBVariantAccount,
   ImportableHDAccount,
 } from './types/account';
+import {
+  BlockTransactionWithLogEvents,
```
Comment (taimanhui) [inline packages/engine/src/index.ts]: lots of 'no-used-vars' warnings

File: packages/engine/src/index.ts [hunk 2]
```diff
@@ -1866,7 +1873,12 @@ class Engine {
       return getTxHistories(chainId, dbAccount.address, pageNumber, pageSize);
     }
 
-    const localHistory = await this.getHistory(networkId, accountId);
+    const localHistory = await this.getHistory(
+      networkId,
+      accountId,
+      undefined,
+      true,
+    );
```
Comment (taimanhui) [inline packages/engine/src/index.ts:1739]: nothing changes effectively

File: packages/engine/src/index.ts [hunk 3]
```diff
@@ -1879,24 +1891,52 @@ class Engine {
...
+          hasMore: false,
+          pageNumber: 1,
+          pageSize: 1,
+          totalCount: 1,
+        },
+        items: [],
+        txList: [],
+      },
+    };
+    let txList = result.data.txList ?? [];
```
Comment (taimanhui) [inline packages/engine/src/index.ts:1772]: remove the `?? []` part as `result.data.txList` is always an empty array.

File: packages/engine/src/proxy.ts [hunk 4]
```diff
@@ -159,9 +159,9 @@ export function fillUnsignedTxObj({
 
   const { type, nonce, feeLimit, feePricePerUnit, ...payload } = extra as {
     type: string;
-    nonce: number;
-    feeLimit: BigNumber;
-    feePricePerUnit: BigNumber;
+    nonce?: number;
```
Comment (taimanhui) [inline packages/engine/src/proxy.ts]: isn't `type` also optional?

File: packages/engine/src/vaults/impl/evm/VaultHelper.ts [hunk 5]
```diff
@@ -58,4 +59,31 @@ export default class VaultHelper extends VaultHelperBase {
...
+
+    if (!nativeTx) {
+      console.log('nativeTx is null');
+      return null;
+    }
+
+    return {
+      ...nativeTx,
+      from: nativeTx.from ?? '',
+      to: nativeTx.to ?? '',
```
Comment (taimanhui) [inline packages/engine/src/vaults/impl/evm/VaultHelper.ts:86]: use lowercase for addresses

File: packages/engine/src/index.ts (hunk 3 above)
Comment (sidmorizon) [reply packages/engine/src/index.ts:1772]: 这里逻辑是这样的，交易历史默认是空数组，之前的逻辑，如果 covalent api 发生异常，导致本地交易也无法返回，因为整个逻辑混在一个方法里了，后期是要把 covalent 的逻辑移除掉

**Review by taimanhui — APPROVED** [review]

**Review by sidmorizon — APPROVED** [review]

**Review by ByteZhang1024 — APPROVED** [review]

**Pull request merged**